pkg/correlation/README.md

```diff
@@ -0,0 +1,103 @@
+## Correlation logging instructions
+
+There are three ways for using this package. All methods will result in a correlation ID being 
+added to the log lines. Each method may make sense in different scenarios. We've provided
+all three methods as they can be used interchangeably depending on the developers preference. 
+
+Currently, a correlation context is created as part of a gRPC interceptor for the following openstorage packages:
+- The CSI Driver (/csi)
+- The SDK Server (/api/server/sdk)
+
+### Method 1: Per-package correlation logger
+This method is nice if a package maintainer wants to register a global 
+logging hook at a per-package level. This is nice as you only need to
+create the logger once per package. However, each log line can be quite
+verbose as you must provide WithContext(ctx)
+
+File 1:
+```
+package example
+
+var (
+    clogger := correlation.NewPackageLogger("test")
+)
+	
+func testFunc() {
+    ctx := correlation.NewContext(context.Background(), "source-component")
+
+    clogger.WithContext(ctx).Info("test info log 1")
+    testFuncTwo(ctx)
+}
+```
+
+File 2:
+```
+package example
+
+func testFuncTwo(ctx context.Context) {
+    clogger.WithContext(ctx).Info("test info log 2")
+}
+```
+
+### Method 2: Per-function correlation logger
+This method is great for reducing the amount of count needed per-line, 
+as you do not need to pass in the context at every log line. However,
+you must remember to instantiate the logger at every function.
+
+```
+package example
+
+func testFunc() {
+    ctx := correlation.NewContext(context.Background(), "test_origin")
```

Comment:
Suggestion

```go
package example

func testFunc(ctx context.Context) {
    ctx = correlation.NewContext(ctx, "example")
    clogger := correlation.NewFunctionLogger(ctx)
    clogger.Info("test info log 1")
    testFuncTwo(ctx)
}
```

Also, what happens if we just pass `ctx` without adding the `NewContext()` call right before it?

Reply:
I think the existing example is sufficient. `testFunc` creates the context from no context at all. 

testFuncTwo logs will include this context.